Libraries/MiKTeX/Core/Session/init.cpp

```diff
@@ -952,7 +954,16 @@ void SessionImpl::SetEnvironmentVariables()
     }
   }
   MIKTEX_ASSERT(!gsDirectories.Empty());
+
+#if MIKTEX_WINDOWS
   Utils::SetEnvironmentString("MIKTEX_GS_LIB", StringUtil::Flatten(gsDirectories, PathName::PathNameDelimiter));
+#else
+  string origGsDirs;
+  if (Utils::GetEnvironmentString("GS_LIB", origGsDirs))
+  {
+    gsDirectories.push_back(origGsDirs);
```

Comment:
That's not correct because `GS_LIB` contains a vector of strings (separated by `:`).

Reply:
> That's not correct because `GS_LIB` contains a vector of strings (separated by `:`).

True, but `GetEnvironmentString` will return only one string, having all paths listed in original `GS_LIB` separated by `:`, and that single string can be added as such into `gsDirectories` vector.

I can make the implementation better and concise, rough sketch as follows:
```
string origGsLib;
if (Utils::GetEnvironmentString("GS_LIB", origGsLib))
{
  vector<string> origGsDirs = StringUtil::Split(origGsLib, PathName::PathNameDelimiter);
  gsDirectories.insert(gsDirectories.end(), origGsDirs.begin(), origGsDirs.end());
}
```

Are you OK with this approach?

Reply:
Looks better. Should we also check each directory exists? I leave this decision up to you.

Reply:
You mean each directory already present in `GS_LIB` (those that will be prepended are already checked with your recent changes and moving `#endif` directive above)?

I do not find that necessary - let us keep whatever is somehow set. Nevertheless, my fresh Ubuntu VMs show that `GS_LIB` is not even set. Even if it was, prepending MiKTeX's paths would do (original `GS_LIB` paths will be eventually used only as fallback).